### PR TITLE
Add angular wrapper for the Field Filter controls

### LINKED ISSUE #95: Add angular wrapper for the Field Filter controls
**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is.
The field filter controls exist, but not the angular wrapper. 

**Additional context**
I'd like a wrapper for them to be able to use the field filter controls in my angular app.

I've already have code for this, and fixed an issue in datepicker where it wasn't allowing the range configuration.

### PR DESCRIPTION
**Pull request by Sovia**
> Explain the **details** for making this change. What existing problem does the pull request solve?
Adding an angular wrapper for the Field Filter controls, and fixing the date picker to be able to handle date ranges.

> **Related issue (required)**:
[https://github.com/infor-design/enterprise-ng/issues/95](https://github.com/infor-design/enterprise-ng/issues/95)

> Other Details:
`closes #95`

### CODE REVIEW

File: src/soho/datepicker/soho-datepicker-reactive-form.spec.ts [hunk 1]
```diff
@@ -50,7 +50,7 @@ class SohoDatePickerReactiveFormTestComponent {
   }
 }
 
-describe('SohoDatePickerComponent on Reactive Form', () => {
+fdescribe('SohoDatePickerComponent on Reactive Form', () => {
```
Comment (fitzorama) [inline src/soho/datepicker/soho-datepicker-reactive-form.spec.ts]: should be describe?

File: src/app/field-filter/field-filter.demo.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    },
+    dateRange: {
+      value: '',
+      filterType: 'equals'
+    }
+  };
+
+  public filterSettings: SohoFieldFilterSettings = {
+    dataset: [
+    { value: 'equals', text: 'Equals', icon: 'filter-equals' },
```
[the remark this answers is not in the document]
Comment (Sovia) [reply src/app/field-filter/field-filter.demo.ts]: Looking at the controls, I don't see any constraints on what could be in this dataset.

[the remark this answers is not in the document]
Comment (bthharper) [reply src/app/field-filter/field-filter.demo.ts]: Okay - are these just menus then?

[the remark this answers is not in the document]
Comment (tmcconechy) [reply src/app/field-filter/field-filter.demo.ts]: Its not obvious but this is the running list .. http://master-enterprise.demo.design.infor.com/components/icons/example-filter-dropdown.html

Source: https://github.com/infor-design/enterprise/blob/master/app/views/components/icons/example-filter-dropdown.html

[the remark this answers is not in the document]
Comment (Sovia) [reply src/app/field-filter/field-filter.demo.ts]: Thanks @tmcconechy . I've constrained the filter options to the values listed there.

**Pull request merged**